### PR TITLE
Add Ruby implementation

### PR DESCRIPTION
**Pull request by buren**

### CODE REVIEW

File: platform/ruby/mail_checker.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
...prumours.com","toss.pw","totalvista.com","totesmail.com","tp-qa-mail.com","tranceversal.com","trasz.com","trollproject.com","tropicalbass.info","trungtamtoeic.com","ttszuo.xyz","tualias.com","txtadvertise.com","ufacturing.com","uguuchantele.com","uhhu.ru","unimark.org","unit7lahaina.com","uploadnolimit.com","urfunktion.se","utiket.us","uwork4.us","vaati.org","valhalladev.com","verdejo.com","veryday.ch","veryday.eu","veryday.info","victoriantwins.com","vikingsonly.com","vinernet.com","vipxm.net ","vixletdev.com","vmailing.info","vmani.com ","vmpanda.com","vorga.org","votiputox.org","voxelcore.com","wakingupesther.com","watchever.biz","watchironman3onlinefreefullmovie.com","wbml.net","webtrip.ch","welikecookies.com","wg0.com","whatifanalytics.com","whiffles.org","wibblesmith.com","widget.gg","wimsg.com ","wralawfirm.com","x1x.spb.ru","x24.com","xcompress.com","xcpy.com","xjoi.com","xn--9kq967o.com","xrho.com","xwaretech.com","xwaretech.info","xwaretech.net","xww.ro","yaqp.com","ynmrealty.com","yougotgoated.com","youneedmore.info","yourewronghereswhy.com","yourlms.biz","yspend.com","yugasandrika.com","yui.it","zepp.dk","zipsendtest.com","zoetropes.org","zombie-hive.com","zumpul.com"].
```
Comment (cabello) [inline platform/ruby/mail_checker.rb]: If every platform has it's own list that will be hard to maintain and keep it consolidated. Maybe JS, Node, PHP, Ruby, and others should all read from the `list.json` available on the root of the repository?

File: platform/ruby/mail_checker.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  end
+
+  def self.extract_domain(email)
+    domain = email.gsub(/.+@([^.]+)/, '\1').downcase
+    domain.split('.')[-2,2].join('.') # Don't include subdomains
+  end
+end
+
+def MailChecker(email)
```
Comment (buren) [inline platform/ruby/mail_checker.rb]: I don't like the method name being `MailChecker`, meaning that you will call this as `MailChecker('someone@example.com')`, which is anything but idiomatic Ruby.. But I opted to follow the style of the node & PHP implementation for now.

What I rather would like to do is rename `MailCheckerImpl` to `MailChecker` and then call it like this `MailChecker.test('someone@example.com')`.

@FGRibreau any thoughts?

Comment (FGRibreau) [inline platform/ruby/mail_checker.rb]: Indeed, we should change this in the next major upgrade, duly noted!

**Comment by buren** [comment]
Replaced by #34

**Pull request closed without merging**